rest/oidc_api.go

```diff
@@ -99,9 +108,38 @@ func (h *handler) handleOIDCCommon() (redirectURLString string, err error) {
 		return redirectURLString, err
 	}
 
+	if !provider.IsDefault {
+		base.Debugf(base.KeyAuth, "Adding provider (%v) to callback URL", base.UD(provider.Name))
+		if err = addCallbackURLQueryParam(redirectURL, requestParamProvider, provider.Name); err != nil {
+			base.Errorf("Failed to add provider to callback URL, err: %v", err)
+		}
+		base.Debugf(base.KeyAuth, "Callback URL: %s", redirectURL.String())
+	}
+
 	return redirectURL.String(), nil
 }
 
+func addCallbackURLQueryParam(uri *url.URL, name, value string) error {
+	if uri == nil {
+		return errors.New("URL must not be nil")
+	}
+	if name == "" {
+		return errors.New("parameter name must not be empty")
+	}
```

Comment:
Is it valid to have an empty provider name? If not, should we return an error for this scenario?